### PR TITLE
Bridge dynamic 2.0

### PR DESCRIPTION
**Pull request by Tifaifai**
Hello,

After four years (testing/update/real use case), we are back with the latest version of mosquitto (mosquitto 2.0.0) with the Dynamic Bridge and payload support JSON Message to create/delete bridge.

In this PR, we submit a new functionality for the bridge (mosquitto_bridge tool).
Our commit propose to manage bridge dynamically by a publish message on $BRIDGE topic.
This new functionality permit to create and delete bridges after the config file step.

History: Reopen #535 / #639 / #653 / #1811 with 'Enhancement' and 'mosquitto-broker' Labels.
More explainations on this functionnality in documentation or PR #535.
Our use case is IoT management with different brokers on different Embedded Systems.

@ralight , we'd be honored if you'd include this feature in your master branch.

Thank you
Tifaifai
Signed-off-by: Tifaifai Maupiti <tifaifai.maupiti@gmail.com>

Thank you for contributing your time to the Mosquitto project!

Before you go any further, please note that we cannot accept contributions if
you haven't signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php).
If you aren't able to do that, or just don't want to, please describe your bug
fix/feature change in an issue. For simple bug fixes it is can be just as easy
for us to be told about the problem and then go fix it directly.

Then please check the following list of things we ask for in your pull request:

- [x] Have you signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php), using the same email address as you used in your commits?
- [x] Do each of your commits have a "Signed-off-by" line, with the correct email address? Use "git commit -s" to generate this line for you.
- [ ] If you are contributing a new feature, is your work based off the develop branch?
- [ ] If you are contributing a bugfix, is your work based off the fixes branch?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully run `make test` with your changes locally?

-----

### CODE REVIEW

**Comment by ralight** [comment]
This is one of my priorities for 2.1. You have been very patient and I applaud you for it.

What I would like to do though, is migrate what you have done to use $CONTROL and mosquitto_ctrl, and take a look at the json api to get some consistency in approach between what we now have in the repo and your work. How does that sound?

Perhaps a good approach would be for me to accept this into a new, temporary branch, and then I can go and tweak to my hearts desire and you don't have an extra burden.

**Comment by ralight** [comment]
One point is that because of the size of your contribution we need to go through slightly more involved Eclipse IP procedures, it isn't just as simple as me accepting the PR.

**Comment by Tifaifai** [comment]
We have moved from $SYS to $BRIDGE to correctly identify the functionality for the bridge. Also in agreement with $CONTROL.

Your proposal is a good idea and approach ;)

**Comment by ralight** [comment]
🎺🎺🎺!🎺

**Comment by zpfvo** [comment]
Congratulations!

**Comment by Deishelon** [comment]
Hi, any plans to release this feature?

File: client/client_shared.h [hunk 1]
```diff
@@ -20,6 +20,7 @@ SPDX-License-Identifier: EPL-2.0 OR EDL-1.0
 #define CLIENT_CONFIG_H
 
 #include <stdio.h>
+#include <mosquitto_broker_internal.h>
```
Comment (Deishelon) [inline client/client_shared.h:23]: This change breaks the cmake build. 

```
[ 29%] Built target mosquittopp
Scanning dependencies of target mosquitto_rr
[ 30%] Building C object client/CMakeFiles/mosquitto_rr.dir/rr_client.c.o
In file included from /home/nikita/projects/ext/mosquitto/client/rr_client.c:38:
/home/nikita/projects/ext/mosquitto/client/client_shared.h:23:10: fatal error: 'mosquitto_broker_internal.h' file not found
#include <mosquitto_broker_internal.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [client/CMakeFiles/mosquitto_rr.dir/build.make:63: client/CMakeFiles/mosquitto_rr.dir/rr_client.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:317: client/CMakeFiles/mosquitto_rr.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

```

**Comment by virajkanwade** [comment]
When will this be available? Almost 9 months!

**Comment by honglei** [comment]
So，it is still not include in release version?

**Pull request merged**